### PR TITLE
Improve create method return value by considering as_module args

### PR DESCRIPTION
**Pull request by msudgh**
Hey, I'm using `python-gist` as an external module, actually, import `GistAPI` in my application but the return value of methods aren't useful in this way, I decided to add an argument with name `as_module` to `GistAPI` class to indicate the way of using of library. By default, the value of this arg is `False` and will not affect the previous process just if `as_module` was `True` it will return the `response` context.

Also, I'm not sure about versioning, `minor` should be increased?

### CODE REVIEW

**Comment by jdowner** [comment]
HI @msudgh. Thanks for making these changes. I am not going to accept the PR as it stands. There are a few things that I am going to add comments on so that you know what I am looking for in a PR, but I would like to better understand the problem you are trying to solve because I don't like the ```as_module``` solution as it stands.

The first thing I would recommend is that you rebase the changes from master. I should not see my changes in this changeset.

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
...
-simplejson
+certifi==2018.4.16
+chardet==3.0.4
+docopt==0.6.2
+gnupg==2.3.1
+idna==2.6
+psutil==5.4.5
+requests==2.18.4
+simplejson==3.14.0
+urllib3==1.22
```
Comment (jdowner) [inline requirements.txt]: Please revert this change. I do not want the additional package requirements here, and I do not want the requirements frozen at a specific revision.

File: bin/gist [hunk 2]
```diff
@@ -134,6 +134,8 @@ import termios
 import docopt
 import gnupg
 import simplejson as json
+import sys
+sys.path.append('./')
```
Comment (jdowner) [inline bin/gist]: This should not be necessary and it is not a good idea to modify ```sys.path```.

File: .idea/vcs.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jdowner) [inline .idea/vcs.xml]: I see another commit that removes this file. Make sure that you squash those commits because this accident should not be preserved in the repo history.

File: gist/gist.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 import os
 import re
 import requests
+from requests import HTTPError
```
Comment (jdowner) [inline gist/gist.py]: Please do not strip the package name from the class.

File: .travis.yml [hunk 5]
```diff
@@ -9,7 +9,7 @@ python:
 # command to install dependencies
 install:
   - "pip install -r requirements.txt"
-  - "pip install pycodestyle responses"
+  - "pip install pep8 responses"
```
Comment (jdowner) [inline .travis.yml]: Revert this changeset.

**Comment by msudgh** [comment]
Ok, let me clear my PR and after that, I'll make a new PR.
Thanks

File: gist/gist.py [hunk 6]
```diff
@@ -128,16 +129,18 @@ class GistAPI(object):
     This class defines the interface to github.
     """
 
-    def __init__(self, token, editor=None):
+    def __init__(self, token, editor=None, as_module=False):
```
Comment (jdowner) [inline gist/gist.py]: Added an argument that specifies how the class is being used is not a solution that I like. The class should behave the same in any environment. Create a ticket that describes the problem you are having here, and together we should be able to figure out an acceptable solution.
